internal/plugin/grpc_provider.go

```diff
@@ -129,23 +128,33 @@ func (p *GRPCProvider) GetProviderSchema() providers.GetProviderSchemaResponse {
 		return resp
 	}
 
-	resp.Provider = convert.ProtoToProviderSchema(protoResp.Provider)
+	identResp, err := p.client.GetResourceIdentitySchemas(p.ctx, new(proto.GetResourceIdentitySchemas_Request))
+	if err != nil {
```

Comment:
We want to check the unimplemented status here too, especially since this is going to be the primary call point from core for now. Or maybe this should just our own `GetResourceIdentitySchemas` method instead?

Reply:
How should Terraform handle `GetResourceIdentitySchemas` errors here?

1. Always continue on error and return at least the provider schema (current implementation)
2. Only continue in case of an unimplemented status and return the provider schema; any other `GetResourceIdentitySchemas` error will be raised.

I assumed we want to do the first case, because `GetProviderSchema` can still return something useful even if the resource identity schema calls fail

Reply:
I think we want `#2`. If we call `GetResourceIdentitySchemas` and there is an another error type, that means there is an actual problem with the provider and we want to raise that somewhere. Having a missing identity schema that needs to be used would of course cause errors down the line, but then we're left debugging how that schema was lost when we should have caught the error at the moment the call was made.

Reply:
Gotcha! Appreciate the additional context